### PR TITLE
[SDK-2970] Remove captcha for enterprise SSO connections

### PR DESCRIPTION
**Pull request by stevehobbsdev**
### Changes

This PR hides the Captcha for enterprise (non-ADFS) SSO connections. As the user is redirected to another vendor to input their password, having a Captcha here makes no sense and is not validated by Auth0 server anyway. However, leaving it in place is an awkward experience, as it looks like the Captcha can be bypassed (you can enter a random value and it still works).

This PR also rolls in #2065, which has now been closed.

### References

`SDK-2970` (internal support ticket)

### Testing

- [x] This change adds unit test coverage
- [ ] This change adds integration test coverage
- [x] This change has been tested on the latest version of the platform/language

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [x] All code quality tools/guidelines have been run/followed
- [x] All relevant assets have been compiled

### CODE REVIEW

File: src/connection/database/login_pane.jsx [hunk 1]
```diff
@@ -57,7 +61,9 @@ export default class LoginPane extends React.Component {
       );
 
     const captchaPane =
-      l.captcha(lock) && l.captcha(lock).get('required') ? (
+      l.captcha(lock) &&
+      l.captcha(lock).get('required') &&
+      (isHRDDomain(lock, databaseUsernameValue(lock)) || !sso) ? (
```
Comment (stevehobbsdev) [inline src/connection/database/login_pane.jsx:66]: This is the real crux of the change - showing captcha if required, but also for ADFS SSO connections only (other SSO connections should hide the Captcha).

File: src/__tests__/engine/classic/sign_up_pane.test.jsx [hunk 2]
```diff
@@ -44,6 +56,50 @@ describe('SignUpPane', () => {
...
+    });
+
+    require('engine/classic').isSSOEnabled.mockReturnValue(true);
+
+    const Component = getComponent();
+
+    expectShallowComponent(<Component {...defaultProps} />).toMatchSnapshot();
+  });
+
+  it('shows the Captcha pane for SSO (ADFS) connections', () => {
```
Comment (jfromaniello) [inline src/__tests__/engine/classic/sign_up_pane.test.jsx:88]: Another one that goes thru the same endpoint and is enterprise is the "ad" strategy... Are we also handling that case?

Comment (stevehobbsdev) [reply src/__tests__/engine/classic/sign_up_pane.test.jsx:88]: The logic here relies on the result of `isHRDDomain`, which looks like it checks both the `ad` and `auth0-adldap` strategies:

https://github.com/auth0/lock/blob/c692eb638543d4b85a740b19fd639ec861ef37b4/src/connection/enterprise.js#L142

Is this sufficient?

Comment (jfromaniello) [reply src/__tests__/engine/classic/sign_up_pane.test.jsx:88]: mak sense!

**Review by adamjmcgrath — APPROVED** [review]

**Comment by sergioaguirreok** [comment]
Hi team, Mercado Libre is evaluating Bot Detection, their use case is B2E, and they have one AD connection to validate users. Does Bot Detection support for non Auth0 DB connections? They are using lock.js, and we tried with the default lock.js template, and it didn’t work with an AD connection. Is that related to this change?
image (1)
image (2)
image
Thanks!

**Comment by thameera** [comment]
Looks like we show the captcha only when a DB connection is enabled as well. Opened ESD-17783.

**Pull request merged**